auth/src/krb.cpp

```diff
@@ -0,0 +1,942 @@
+        if (my_creds == GSS_C_NO_CREDENTIAL) {
+            majorStatus = gss_acquire_cred(
+                    &minorStatus,
+                    myName,
+                    0,
+                    GSS_C_NULL_OID_SET,
+                    _specified_name.empty() ? GSS_C_INITIATE : GSS_C_ACCEPT,
+                    &my_creds,
+                    NULL,
+                    NULL);
+        }
+        else {
+            // Already have credentials so all is well
+            majorStatus = GSS_S_COMPLETE;
+        }
+
+        if (majorStatus != GSS_S_COMPLETE) {
+            static const char* msg = "Failed acquiring credentials.";
+            krb_log_error(_r_error, msg, majorStatus, minorStatus, _is_client);
+            THROW(KRB_ERROR_ACQUIRING_CREDS, msg);
+        }
+
+        // set the credentials in the auth object
+        gss_release_name(&minorStatus, &myName);
+
+        majorStatus = gss_inquire_cred(&minorStatus, my_creds, &myName2, NULL, NULL, NULL);
+        if (majorStatus != GSS_S_COMPLETE) {
+            static const char* msg = "Failed inquiring creds for the name.";
+            krb_log_error(_r_error, msg, majorStatus, minorStatus, _is_client);
+            THROW(KRB_ERROR_ACQUIRING_CREDS, msg);
+        }
+
+        gss_OID doid2;
+        gss_buffer_desc client_name2;
+        majorStatus = gss_display_name(&minorStatus, myName2, &client_name2, &doid2);
+        if (majorStatus != GSS_S_COMPLETE) {
+            static const char* msg = "Failed during displaying name.";
+            krb_log_error(_r_error, msg, majorStatus, minorStatus, _is_client);
+            THROW(KRB_ERROR_DISPLAYING_NAME, msg);
+        }
+
+        std::string display_name;
+        if (client_name2.value && client_name2.length > 0) {
+            display_name = std::string{static_cast<char*>(client_name2.value)};
+        }
+
+        majorStatus = gss_release_name(&minorStatus, &myName2);
+        if (majorStatus != GSS_S_COMPLETE) {
+            static const char* msg = "Failed to release cred name.";
+            krb_log_error(_r_error, msg, majorStatus, minorStatus, _is_client);
+            THROW(KRB_ERROR_RELEASING_NAME, msg);
+        }
+
+        // TODO: should this be released?
+        gss_release_buffer(&minorStatus, &client_name2);
+
+        return display_name;
+    } // krb_setup_creds
+
+    /// @brief Import the specified name into a KRB name
+    void krb_import_name(rError_t* _r_error,
+                         const char* _service_name,
+                         gss_name_t* _output_name,
+                         bool _is_client)
+    {
+        //gss_name_t* imported_name = std::malloc(sizeof(gss_name_t));
+        //std::memset(imported_name, 0, sizeof(gss_name_t));
+        *_output_name = GSS_C_NO_NAME;
+
+        if (!_service_name || std::string_view{_service_name}.empty()) {
+            return;
+        }
+
+        const std::size_t size = std::strlen(_service_name) + 1;
```

Comment:
These lines could be adjusted to possibly avoid duplicating the work to get the string's length.

Again, not a deal breaker.

Reply:
Will do

Reply:
Done